torch_lr_finder/lr_finder.py

```diff
@@ -460,6 +463,19 @@ def plot(self, skip_start=10, skip_end=5, log_lr=True, show_lr=None, ax=None):
 
         # Plot loss as a function of the learning rate
         ax.plot(lrs, losses)
+
+        # Plot the suggested LR in a one-time loop
+        while suggestion:
+            try: 
+                mg = (np.gradient(np.array(losses))).argmin()
+            except:
+                print("Failed to compute the gradients, there might not be enough points.")
+                break
```

Comment:
Also don't see a reason for the try-except block here. In general, if you can't recover from the exception don't catch it, let it fail

Reply:
The `while` and `try`-`except` block was meant to isolate the `suggest_lr` part, so if the `suggest_lr` fail, it won't make the `plot` fail. :sweat_smile:

Reply:
Ah, I see, that makes sense. Then I would replace the `while` with an `if` and leave the rest

Reply:
Emm...this code is quite confusing... I used the `while` loop to skip `plot` the suggested LR (if `np.gradient` fail)...
Maybe this is a better way? :point_down:
```suggestion
            try: 
                mg = (np.gradient(np.array(losses))).argmin()
            except:
                print("Failed to compute the gradients, there might not be enough points.")
                mg = None
            if mg is not None:
                # plot
```

---

Maybe in the future we can implement some functions for different `suggest_lr` algorithms? 
```python
if suggest_lr == 'steepest':
    self._plot_steepest(history, ax)
elif suggest_lr == 'XXX'
    self._plot_XXX(history, ax)
else:
    print ('Oops!')
```